SwiftUIHomeworks/SwiftUIHomeworks/Views/MainView.swift

```diff
@@ -0,0 +1,216 @@
+      }
+      .padding(.horizontal, Constants.playerBarLengthValue)
+      HStack(spacing: Constants.spacingValue) {
+        backSongButton
+        playStopButton
+        nextSongButton
+      }
+      HStack {
+        sharingButton
+        Spacer()
+        saveButton
+      }
+    }
+    .frame(maxHeight: .infinity)
+    .padding(.horizontal, Constants.backgroundLengthValue)
+    .background(backgroundCoverView(coverName: playerViewModel.getCoverName()))
+    .ignoresSafeArea()
+  }
+  
+  // MARK: - Private properties
+  
+  @ObservedObject private var playerViewModel = PlayerViewModel()
+  @State private var isPlaying = true
+  @State private var isShareAlertShown = false
+  @State private var isSaveActionSheetShown = false
+  
+  private var playStopButton: some View {
+    Button(action: {
+      self.isPlaying.toggle()
+      guard self.isPlaying else {
+        self.playerViewModel.play()
+        return
+      }
+      self.playerViewModel.stop()
+    }) {
+      Image(systemName: isPlaying ? Constants.playImageName : Constants.stopImageName)
+        .resizable()
+        .foregroundColor(.black)
+        .aspectRatio(contentMode: .fill)
+        .frame(width: Constants.playWidthValue, height: Constants.playHeightValue)
+    }
+  }
+  
+  private var nextSongButton: some View {
+    Button {
+      playerViewModel.nextSong()
+    } label: {
+      Image(systemName: Constants.forwardImageName)
+        .resizable()
+        .foregroundColor(.black)
+        .aspectRatio(contentMode: .fill)
+        .frame(width: Constants.nextPreviousWidthValue, height: Constants.nextPreviousHeightValue)
+    }
+  }
+  
+  private var backSongButton: some View {
+    Button {
+      playerViewModel.previousSong()
+    } label: {
+      Image(systemName: Constants.backwardImageName)
+        .resizable()
+        .foregroundColor(.black)
+        .aspectRatio(contentMode: .fill)
+        .frame(width: Constants.nextPreviousWidthValue, height: Constants.nextPreviousHeightValue)
+    }
+  }
+  
+  private var songImage: some View {
+    Image(playerViewModel.getCoverName())
+      .resizable()
+      .frame(maxWidth: UIScreen.main.bounds.width - Constants.oneHundredValue, maxHeight: UIScreen.main.bounds.height / Constants.threeValue)
+      .aspectRatio(contentMode: .fill)
+      .cornerRadius(Constants.cornerRadiusValue)
+  }
+  
+  private var artistText: some View {
+    Text(playerViewModel.getArtist())
+      .font(.title2)
+      .bold()
+      .padding()
+  }
+  
+  private var nameText: some View {
+    Text(playerViewModel.getName())
+      .padding()
+  }
+  
+  private var playerSlider: some View {
+    Slider(value: Binding(get: {
+      playerViewModel.durationSong
+    }, set: { newValue in
+      playerViewModel.durationSong = newValue
+      playerViewModel.setSongDuration(durationValue: playerViewModel.durationSong)
+    }), in: Constants.zeroValue...playerViewModel.maxDuration)
+  }
+  
+  private var currentTimeText: some View {
```

Comment:
```suggestion
  private var currentTimeTextView: some View {
```
везде где возвращаем View на конце View